### PR TITLE
🚧 fix #289

### LINKED ISSUE #289: 🐛 `-` dash instances are omitted when `replacement` is set to `_` underscore.
When providing the replacement as `_` the `-` instances are wrongfully omitted.

```js
slug('arabic-english translator') // "arabic-english-translator"
slug('arabic-english translator', { replacement: '_' }) // "arabicenglish_translator" (this is wrong, should be "arabic-english_translator")
```

expected behavior:
```js
slug('arabic-english translator', { replacement: '_' }) // "arabic_english_translator"
```

### PR DESCRIPTION
**Pull request by A-Tokyo**
Closes https://github.com/Trott/slug/issues/289

### CODE REVIEW

File: test/slug.test.js [hunk 1]
```diff
@@ -50,9 +50,10 @@ describe('slug', function () {
     assert.throws(slug, /slug\(\) requires a string argument/)
   })
 
-  it('should replace whitespaces with replacement', function () {
+  it('should replace whitespaces with replacement and preserve default replacement', function () {
     assert.strictEqual(slug('foo bar baz'), 'foo-bar-baz')
     assert.strictEqual(slug('foo bar baz', '_'), 'foo_bar_baz')
+    assert.strictEqual(slug('foo-bar baz', '_'), 'foo-bar_baz')
```
Comment (Trott) [inline test/slug.test.js:56]: I don't think this is right. If you specify that you want your lone non-alphanumeric character to be `_`, then treating `-` as a preserved character is a bug. (We don't preserve `_`.) I could see *maybe* arguing that certain symbols (`-`, `_`, and a handful of others) be converted to the replacement character along with spaces. But preserving them seems like a bug to me.

Comment (A-Tokyo) [reply test/slug.test.js:56]: @Trott You are right!

I believe I meant to write :

```
    assert.strictEqual(slug('foo-bar baz', '_'), 'foo_bar_baz')
```

Comment (A-Tokyo) [inline test/slug.test.js:56]:
```suggestion
    assert.strictEqual(slug('foo-bar baz', '_'), 'foo_bar_baz')
```

Comment (Trott) [reply test/slug.test.js]: `slug()` explicitly does not preserve punctuation/symbols by default. `slugify()` does.

```console
> slug('foo-bar_baz')
'foo-barbaz'
> slugify('foo-bar_baz')
'foo-bar_baz'
>
```

If you want slug to treat convert `-` to `_`, you need to add it to the charmap.

```console
> slug.charmap['-'] = '_'
'-'
> slug('foo-bar baz', {replacement: '_'})
'foo_bar_baz'
> 
```

Comment (A-Tokyo) [reply test/slug.test.js]: @Trott Thanks for your response ( :

I think I'll use this solution

**Pull request closed without merging**